modules/core/src/lib/layer.js

```diff
@@ -804,7 +820,10 @@ ${flags.viewportChanged ? 'viewport' : ''}\
   _updateBaseUniforms() {
     const uniforms = {
       // apply gamma to opacity to make it visually "linear"
-      opacity: Math.pow(this.props.opacity, 1 / 2.2)
+      opacity:
+        typeof this.props.opacity === 'function'
+          ? props => Math.pow(this.props.opacity(props), 1 / 2.2)
```

Comment:
Naming the argument `props` is confusing in this context. Add a comment about this being the luma.gl animation callback?

Reply:
> Naming the argument props is confusing in this context

Yes good point. I think they could just be called `animationProps`, I think that is consistent with other use and will make things clear. Will fix.